packages/core/src/modules/company/company-event-service.ts

```diff
@@ -0,0 +1,30 @@
+import { Company, Event } from "@dotkomonline/types"
+import { CompanyEventRepository } from "./company-event-repository" // Note: you might need to create or rename this file based on previous changes.
+import { Cursor } from "../../utils/db-utils"
+
+export interface CompanyEventService {
+  createEvent: (company: Company["id"], event: Event["id"]) => Promise<void>
+  deleteEvent: (company: Company["id"], event: Event["id"]) => Promise<void>
+  getEventsByCompanyId: (company: Company["id"], take: number, cursor?: Cursor) => Promise<Event[]>
+}
+
+export class CompanyEventServiceImpl implements CompanyEventService {
+  constructor(private readonly companyEventRepository: CompanyEventRepository) {}
+
+  async createEvent(company: Company["id"], event: Event["id"]) {
```

Comment:
Because TS sucks at inferring in class overrides, we need explicit return types here, probably why you had to assert in the query